lib/mfg.c

```diff
@@ -226,3 +273,70 @@ int switchtec_active_image_index_set(struct switchtec_dev *dev,
 			    sizeof(set), NULL, 0);
 	return ret;
 }
+
+/**
+ * @brief Read security settings from config file
+ * @param[in]  setting_file	Security setting file
+ * @param[out] set		Security settings
+ * @return 0 on success, error code on failure
+ */
+enum switchtec_parse_setting_error
+switchtec_read_sec_cfg_file(FILE *setting_file,
+			    struct switchtec_security_cfg_set *set)
+{
+	ssize_t rlen;
+	char magic[4] = {'S', 'S', 'F', 'F'};
+	uint32_t crc;
+	struct setting_file_header {
+		uint8_t magic[4];
+		uint32_t version;
+		uint32_t rsvd;
+		uint32_t crc;
+	};
+	struct setting_file_data {
+		uint64_t cfg;
+		uint32_t pub_key_exponent;
+		uint8_t rsvd[36];
+	};
+	struct setting_file {
+		struct setting_file_header header;
+		struct setting_file_data data;
+	} file_data;
+
+	rlen = fread(&file_data, 1, sizeof(file_data), setting_file);
+
+	if (rlen < sizeof(file_data))
+		return SWITCHTEC_SETTING_FILE_ERROR_LEN;
```

Comment:
In this utility context, we use return value and errno for the error returning. Generally, we have two types of errors, system error and mrpc command errror. The system error is like what you are handling here, some check failure or system API failure, please use the standard errno values (like ENOMEM, EINVAL) for them instead of creating function specific ones; mrpc command error is what we get from the MRPC execution. For more detail, pls check this PR #132.